solidity/test/InterchainAccountRouter.t.sol

```diff
@@ -67,29 +69,52 @@ contract InterchainAccountRouterTest is Test {
 
     Callable target;
 
+    function deployProxiedIcaRouter(
+        uint32 _domain,
+        MockMailbox _mailbox,
+        IInterchainGasPaymaster _igps,
+        IInterchainSecurityModule _ism,
+        address _owner
+    ) public returns (InterchainAccountRouter) {
+        InterchainAccountRouter implementation = new InterchainAccountRouter(
+            _domain
+        );
+
+        TransparentUpgradeableProxy proxy = new TransparentUpgradeableProxy(
```

Comment:
this is good for tests but if we want consistent ICA addresses on each chain we have to get a bit clever with CREATE2

Reply:
I'm not sure I follow, as far as I've seen ICA Router addresses and ICA addresses are not consistent across chains.

Reply:
nevermind, we actually already made this decision to move away from consistent addresses, I just forgot